app/controllers/admin/variant_overrides_controller.rb

```diff
@@ -6,30 +6,41 @@ class VariantOverridesController < ResourceController
 
     before_filter :load_spree_api_key, only: :index
     before_filter :load_data
+    before_filter :load_collection, only: [:bulk_update, :bulk_reset]
 
     def index
     end
 
 
     def bulk_update
-      collection_hash = Hash[params[:variant_overrides].each_with_index.map { |vo, i| [i, vo] }]
-      vo_set = VariantOverrideSet.new @variant_overrides, collection_attributes: collection_hash
-
       # Ensure we're authorised to update all variant overrides
-      vo_set.collection.each { |vo| authorize! :update, vo }
+      @vo_set.collection.each { |vo| authorize! :update, vo }
 
-      if vo_set.save
+      if @vo_set.save
         # Return saved VOs with IDs
-        render json: vo_set.collection, each_serializer: Api::Admin::VariantOverrideSerializer
+        render json: @vo_set.collection, each_serializer: Api::Admin::VariantOverrideSerializer
       else
-        if vo_set.errors.present?
-          render json: { errors: vo_set.errors }, status: 400
+        if @vo_set.errors.present?
+          render json: { errors: @vo_set.errors }, status: 400
         else
           render nothing: true, status: 500
         end
       end
     end
 
+    def bulk_reset
+      # Ensure we're authorised to update all variant overrides.
+      @vo_set.collection.each { |vo| authorize! :bulk_reset, vo }
+
+      # Changed this to use class method instead, to ensure the value in the database is used to reset and not a dirty passed-in value
+      #vo_set.collection.map! { |vo| vo = vo.reset_stock! }
+      @vo_set.collection.map! { |vo| VariantOverride.reset_stock!(vo.hub,vo.variant) }
+      render json: @vo_set.collection, each_serializer: Api::Admin::VariantOverrideSerializer
+      if @vo_set.errors.present?
+        render json: { errors: @vo_set.errors }, status: 400
```

Comment:
Does this work? As far as I know, Rails won't let you render twice within the one request, so by the time we hit this render call for the errors, we have already rendered the collection, and I expect we'll get an exception.